### DISCUSSION TITLE
How do I apply a meta tag to all layouts? PWA Pinch to Zoom Must be Disabled.

### DISCUSSION THREAD

**Discussion by Inzendis** [vercel/next.js#56554] [Help]
### Summary

My root layout which contains the html tag has (Using App Router)

`<meta
          name="viewport"
          content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0, height=device-height viewport-fit=cover"
        />
        <meta name="mobile-web-app-capable" content="yes" />
        <meta name="apple-mobile-web-app-capable" content="yes" />
        <meta name="HandheldFriendly" content="true" />
        <meta name="msapplication-TileColor" content="#1f2023" />
        <meta name="theme-color" content="#1f2023"></meta>`
        
        
It works as inteded for my PWA app but when I navigate to other pages the pinch to zoom is enabled again. When I close the pwa and open it again. I made it so the if there a valid token cookie still in place then it redircts to home page (or dashboard), in this case the pinch is disabled (as I want it) its just not applying to the other pages.
 
 I have nested layouts done like the docs say to.
Apperently the docs say that everything in the root layout applied to the inner layouts, meaning <head> tags apply to all pages?

### Example

````
`import React from "react";
export const metadata = {
  title: "App", description: "",
};

export default async function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  const data = await getServerSession(options);
  return (
    <html lang="en">
      <head>
        <title>App</title>
        <meta
          name="viewport"
          content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0, height=device-height viewport-fit=cover"
        />
        <meta name="mobile-web-app-capable" content="yes" />
        <meta name="apple-mobile-web-app-capable" content="yes" />
        <meta name="HandheldFriendly" content="true" />
        <meta name="msapplication-TileColor" content="#1f2023" />
        <meta name="theme-color" content="#1f2023"></meta>
        <link
          href="/splashscreens/iPhone_13_mini__iPhone_12_mini__iPhone_11_Pro__iPhone_XS__iPhone_X_portrait.png"
          media="(device-width: 375px) and (device-height: 812px) and (-webkit-device-pixel-ratio: 3)"
          rel="apple-touch-startup-image"  />
        <link
          href="/splashscreens/iPhone_11__iPhone_XR_portrait.png"
          media="(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 2)"
          rel="apple-touch-startup-image"
        /> <link
          href="/splashscreens/iPhone_11_Pro_Max__iPhone_XS_Max_portrait.png"
          media="(device-width: 414px) and (device-height: 896px) and (-webkit-device-pixel-ratio: 3)"
          rel="apple-touch-startup-image"  />
        <link
          href="/splashscreens/iPhone_14_Plus__iPhone_13_Pro_Max__iPhone_12_Pro_Max_portrait.png"
          media="(device-width: 428px) and (device-height: 926px) and (-webkit-device-pixel-ratio: 3)"
          rel="apple-touch-startup-image" />
        {/* Temporarily disabling OneSignal due to that it can be initialized once, on dev it does it twice. */}
        {/* {data && <UseOneSignal />} */}<link rel="manifest" href="/manifest.webmanifest" />
      </head>
      <body className={inter.className}>{children}</body>
    </html>
  );
}
````

**Comment by s0h311** [comment]
Hi @Inzendis could you provide us your whole `layout.jsx` or `layout.tsx` file.

**Reply by Inzendis** [reply]
Added to the example.

As a fix I might have to add the meta tag on all layours or pages. Its no being passed down from root layout.

**Reply by s0h311** [reply]
@Inzendis take a look at this example

```tsx
import './globals.css'
import type { Metadata } from 'next'
import { Kanit } from 'next/font/google'
import { TrpcProvider } from '@/trpc/trpcProvider'
import { Analytics } from '@vercel/analytics/react'

const kanit = Kanit({  // FONT
  weight: '400',
  style: 'normal',
  subsets: ['latin'],
})

export const metadata: Metadata = {
  title: 'BlaBla',
  description: BlaBla',
}

type RootLayoutProps = {
  children: React.ReactNode
}

export default function RootLayout({ children }: RootLayoutProps) {
  return (
    <TrpcProvider>
      <html lang='en'>
        <head>
          {
            // YOUR META TAGS GO HERE
          }
        </head>
        <body className={`${kanit.className}`}>
          {children}
          <Analytics />
        </body>
      </html>
    </TrpcProvider>
  )
}
```

And it's also mentioned here: https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#modifying-head

**Reply by Inzendis** [reply]
Ty for the reply, isn't my code similar? I got all the meta tags inside the <head> tags. I can't use <head> in nested layouts unless I do a root layout for each page, and that is not the idea.

They do work, its the that specific meta tag that isn't working. Because there is a meta tag to change the status bar color and it works throughout the entire PWA.

Just now tested to put that metatag in each layout and it doesn't work, but I if put it in the page.tsx it works as I want it. But I think this is something that layout should do.

Potential bug?
